src/ControlSystem/ApparentHorizons/Measurements.hpp

```diff
@@ -0,0 +1,121 @@
+#include "ControlSystem/Protocols/Submeasurement.hpp"
+#include "ControlSystem/RunCallbacks.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/LinkedMessageQueue.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "ParallelAlgorithms/Interpolation/Callbacks/ErrorOnFailedApparentHorizon.hpp"
+#include "ParallelAlgorithms/Interpolation/Callbacks/FindApparentHorizon.hpp"
+#include "ParallelAlgorithms/Interpolation/Interpolate.hpp"
+#include "ParallelAlgorithms/Interpolation/Targets/ApparentHorizon.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t VolumeDim>
+class ElementId;
+template <size_t Dim>
+class Mesh;
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+namespace domain::Tags {
+template <size_t Dim>
+struct Mesh;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace control_system::ah {
+struct BothHorizons : tt::ConformsTo<protocols::Measurement> {
+  template <::ah::ObjectLabel Horizon>
+  struct FindHorizon : tt::ConformsTo<protocols::Submeasurement> {
+   private:
+    template <typename ControlSystems>
+    struct Interpolator {
+      static std::string name() {
+        return "ControlSystem::BothHorizons::Ah" + ::ah::name(Horizon);
+      }
+
+      struct temporal_id : db::SimpleTag {
+        using type = LinkedMessageId<double>;
+      };
+
+      using vars_to_interpolate_to_target =
+          tmpl::list<gr::Tags::SpatialMetric<3, ::Frame::Grid, DataVector>,
+                     gr::Tags::InverseSpatialMetric<3, ::Frame::Grid>,
+                     gr::Tags::ExtrinsicCurvature<3, ::Frame::Grid>,
+                     gr::Tags::SpatialChristoffelSecondKind<3, ::Frame::Grid>>;
+      using compute_vars_to_interpolate = ::ah::ComputeHorizonVolumeQuantities;
+      using compute_items_on_target = tmpl::list<
+          StrahlkorperGr::Tags::AreaElementCompute<::Frame::Grid>,
+          StrahlkorperTags::ThetaPhiCompute<::Frame::Grid>,
+          StrahlkorperTags::RadiusCompute<::Frame::Grid>,
+          StrahlkorperTags::RhatCompute<::Frame::Grid>,
+          StrahlkorperTags::InvJacobianCompute<::Frame::Grid>,
+          StrahlkorperTags::DxRadiusCompute<::Frame::Grid>,
+          StrahlkorperTags::OneOverOneFormMagnitudeCompute<3, ::Frame::Grid,
+                                                           DataVector>,
+          StrahlkorperTags::NormalOneFormCompute<::Frame::Grid>,
+          StrahlkorperTags::UnitNormalOneFormCompute<::Frame::Grid>,
+          StrahlkorperTags::UnitNormalVectorCompute<::Frame::Grid>,
+          StrahlkorperTags::GradUnitNormalOneFormCompute<::Frame::Grid>,
+          StrahlkorperTags::ExtrinsicCurvatureCompute<::Frame::Grid>,
+          StrahlkorperGr::Tags::SpinFunctionCompute<::Frame::Grid>>;
```

Comment:
Probably don't need this ComputeItem (because nobody is measuring the spin for the control system).

I suspect `StrahlkorperGr::Tags::AreaElementCompute<::Frame::Grid>` isn't used either.